src/main/src/main/java/org/geoserver/catalog/LayerGroupInfo.java

```diff
@@ -198,6 +199,26 @@ public Integer getCode() {
     List<MetadataLinkInfo> getMetadataLinks();
 
     /**
+     * Return the keywords associated with this layer group. If no keywords are available
+     * an empty list should be returned.
+     *
+     * @return a non NULL list containing the keywords associated with this layer group
+     */
+    default List<KeywordInfo> getKeywords() {
+        return Collections.emptyList();
+    }
+
+    /**
+     * Set the keywords of this layer group. The provided keywords will override any existing
+     * keywords no merge will be done.
+     *
+     * @param keywords new keywords of this layer group
+     */
+    default void setKeywords(List<KeywordInfo> keywords) {
```

Comment:
Nope, in the GeoServer catalog lists cannot be set, they are "live" list, you get them and modify them. Setters are often found in Impl classes to help XStream.
Check ResourceInfo:

````
    /**
     * A collection of keywords associated with the resource.
     * 
     * @uml.property name="keywords"
     */
    List<KeywordInfo> getKeywords();

    /**
     * List of keyword values, derived from {@link #getKeywords()}.
     */
    List<String> keywordValues();
````